### ISSUE TITLE
No Functional Media Keys in BM Install

### ISSUE THREAD

**Issue by naclander** [gardotd426/regolith-de#14]
I gave regolith a go, and launched it from lightdm. I was able to open programs, but it seems some of the keybindings were off. Specifically I was not able to quit the session, and I was not able to use the media keys on my laptop to lower the volume, lower the screen brightness, etc.  

I would love to help debug this and improve the AUR regolith package if anyone has any ideas on how to start.

I was using the latest regolith package built from cloning this repo.

**Comment by gardotd426** [comment]
To use media keys on i3, you'll have to set up the keybindings in the config.

**Comment by naclander** [comment]
> To use media keys on i3, you'll have to set up the keybindings in the config.

That can't be right, using Regolith on my machine running Ubuntu I can lower the brightness, volume control, etc, without having to set up anything. The expectation is that the regolith package on Arch would work the same way.

> I'm not sure what you mean by you weren't able to quit the session...

The key-chord to quit the session as posted on the [Regolith website](https://regolith-linux.org/docs/reference/keybindings/#SessionLogout) did not work on my install.

**Comment by gardotd426** [comment]
Ahhh, unfortunately, I've solved the problem. And there's no fixing it. 

Arch and Manjaro recently moved to GNOME 3.38. It doesn't work with regolith. 

Until Regolith itself (meaning until Ubuntu) moves to GNOME 3.38, there's not gonna be anything I can do. 

I'll go ahead and add this to the README that no one should use this until upstream is fixed. I'll go ahead and try and get in touch with the main dev @ Regolith as well, and let him know if he doesn't already. 

I just tested with a machine that wasn't updated (a VM) and still running 3.36, and everything works flawlessly. I updated it, logged out and back in, and sure enough I saw exactly what you saw.

**Comment by gardotd426** [comment]
Well, 20.10 is on 3.38, but Regolith is based on LTS and I don't know if they have a 20.10 build out yet.

**Comment by gardotd426** [comment]
Ah. Here we go:

https://www.omgubuntu.co.uk/2020/11/regolith-linux-1-5-released-based-on-ubuntu-20-10

Seems it was just a few days ago. I'll have to migrate all the packages over, and try to find the repo where they're hosted (it's not on the 1.4 one at least that I can find).

**Comment by naclander** [comment]
I've had this issue for quite a while, so I'm not sure it was just because of the 1.5 update.

**Comment by gardotd426** [comment]
I don't know what to tell you about that, as I and others couldn't reproduce it until now. 

Anyway, go ahead and clone my new testing branch (I mean it's functional, I just haven't updated the checksums and stuff for it to go over to master yet) and try that. It worked fully for me on bare metal. All the gnome-control-center stuff and logging out works again, etc. 

`git clone https://github.com/gardotd426/regolith-de -b 1point5`

You should be able to just reinstall over top the original packages, but you might wanna uninstall them first. Up to you. 

Let me know your results please.

**Comment by naclander** [comment]
I tried creating a new user and running regolith, but I had the same behavior where the media keys are not working. 
They are working fine in Plasma, so my hunch is that it's some global gnome configuration I have on my machine that may be causing issues globally.

I tried reinstalling most gnome packages, but no luck. I think I need to figure out where this gnome is, or reinstall Arch and try again. In the meantime its most likely due to some local configuration on my machine, so you can close this issue. I can create another issue if I rediscover this problem after a re-installation. Thanks for the help.

**Comment by gardotd426** [comment]
Did everything else work though? Like the gnome-control-center keybindings,
the wallpaper, etc.? Like other than the media keys, did it work as
expected?

**Comment by gardotd426** [comment]
Also, go ahead and try pulling the 1point5 branch again, and installing with `makepkg -sfi`.

**Comment by gardotd426** [comment]
Okay, so go ahead and uninstall everything from the PKGBUILD and reinstall after pulling in the changes from the master branch. I've made a bunch of changes, cleaned some stuff up, added a patch to fix the "infinite loading cursor spinny wheel" i3 quirk, and some other stuff, and I just tested it and I have full media key control with my Corsair K70 RGB keyboard, and it's working as intended (meaning that it's the GNOME part being triggered, not some hacky i3 workaround or anything, I get the gnome volume/play/stop icons coming up and everything). Also logout, gnome-control-center, and other misc stuff is working. Please let me know your results.

**Comment by naclander** [comment]
After reinstalling my Arch Linux system I installed regolith from the master branch, and everything seems to work fine with no problems. 

I'm guessing the issues I had earlier were due to some leftover configuration I had on my old system.

**Comment by gardotd426** [comment]
That's fantastic to hear. 

>Seems like this works OK enough to be put on the AUR?

Yeah, I'm getting there. After moving over to 1.5 I needed to change all the sha256sums which for 46 or something packages was something I wasn't looking forward to, but I did it the other day, and that's one of the last things I needed to do. The only real thing left is to run through the repo of .deb packages and make sure there's nothing I need to update. I might add a couple themes too, as I've only included about 5 or 6 of the 8 or 9, and I might add the dracula style and whatnot. But yeah, it's pretty close. I was hoping by now I'd find some people willing to help maintain it, but that hope seems to have died at this point, but no big deal.
